### ISSUE TITLE
Deprecate recursive CompositeView (in v2.3 for removal in v3)

### ISSUE THREAD

**Issue by jmeas** [marionettejs/backbone.marionette#2130]
CompositeViews do two unrelated things.

- They allow you to wrap your collection in a template
- They are recursive

The first thing is very convenient, and I'm sure lots of folks want to use that. The second thing is niche to the point of absurdity. Why was that ever added to the library?

Not only does it make the composite view more difficult to explain, but it creates tough-to-debug errors when you wouldn't expect such strangeness from its default behavior.

Removing it is a fantastic solution. Not only does it resolve all of the aforementioned problems, but it's *dead simple* to add the recursive functionality back in. All that you do is specify the `childView` – like always – but you just reference the view itself instead of leaving it empty.

Another :+1: to this is that we can make the default view a `Marionette.ItemView` – just like the CollectionView – and the overall API of the entire library becomes more consistent.

tl;dr: get outta here, recursive composite view

**Comment by sockmonk** [comment]
I think that a happy middle ground would be to make the CompositeView non-recursive by default, but have either an option to enable it (`recursive: true`) or even subclass a RecursiveView. That makes for fewer surprises for the most common use case of just wrapping a collection, but still lets people get the recursive behavior if they ask for that by name.

**Comment by jmeas** [comment]
@sockmock, the idea is that the recursive functionality is both so rare and so easy to back in *within* your own application that we don't need to do anything else in the core library to support it. We just need to remove it, and everyone's happy.

**Comment by jasonLaster** [comment]
Yea @jmeas agreed that it's rare. with that said, we will HAVE to create a nice official tree view if we take it out. I'll personally be really bummed if I have to build my own because I'm using it in the Inspector. 

Put another way, creating a nice official tree view library will be a good chance to come up with marionette libraries that are designed to be used by others.

**Comment by jmeas** [comment]
@jasonLaster, the beauty of it is that CompositeView  is **still** is that tree class. We're not losing any functionality, so 're-creating it' within your app is as simple as setting the `childView` class. To create the tree view *within* your app, you would simply use your view's constructor as the `childView`.

**Comment by jasonLaster** [comment]
ohhhhhh...... okay

**Comment by abzainuddin** [comment]
+1 Never understood why it was recursive is the default in the first place...

**Comment by paulfalgout** [comment]
This is officially deprecated and so this issue is in process on other tickets
